specs/HostResourceMapping.md

```diff
@@ -0,0 +1,108 @@
+# Background
+We have received feedbacks that there is a need to access local files in WebView2. In this document we describe the APIs for this scenario. We'd appreciate your feedback.
+
+
+# Description
+To access local files in WebView2, define a virutal host name to local folder mapping using the `AddVirtualHostNameToFolderMapping` API,
+and then refer to local files under the folder with a normal http/https url with the virtual host name.
+
+# Examples
+Suppose the app has index.html file next to its exe file. The following code snippet demonstrates how to define a mapping
+for `app-file.invalid` host name so that it can be accessed from pages using normal http/https url like https://app-file.invalid/index.html.
+
+## Win32 C++
+```cpp
+webview2->AddVirtualHostNameToFolderMapping(
+    L"app-file.invalid", L".", COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_DENY_CORS);
+webview2->Navigate("https://app-file.invalid/index.html");
+```
+
+## .Net and WinRT
+```c#
+webView.CoreWebView2.AddVirtualHostNameToFolderMapping(
```

Comment:
What happens if I call `Add` twice for the same host name with the same folder and access policy?

1. Succeeds, bumps refcount. (You must call Remove twice to remove the mapping.)
2. Succeeds, no effect on refcount. Calling Remove removes the mapping immediately.)
3. Fails.

What happens if I call `Add` twice for the same host name with different folder or CORS policy?

1. Overwrites previous mapping, bumps refcount. Each Remove call removes a mapping in LIFO order.
2. Overwrites previous mapping, bumps refcount. Each Remove call removes a mapping in FIFO order.
3. Overwrites previous mapping, bumps refcount. Each Remove call decrements refcount but only most recent Add is in effect.
4. Overwrites previous mapping, no effect on refcount. (You need only call Remove once.)
5. Queues behind previous mapping, takes effect when previous mapping is removed (in FIFO order).
6. Fails.

If the behavior is "Overwrite, no refcount" then maybe this is a "Set" rather than "Add".

* SetVirtualHostNameToFolderMapping / ClearVirtualHostNameToFolderMapping

Reply:
From Liang it sounds like (2) & (4). We can rename to Set and Clear.

Reply:
Updated the API to use Set and Clear